src/alpaka/AlpakaCore/alpakaMemoryHelper.h

```diff
@@ -0,0 +1,44 @@
+#ifndef ALPAKAMEMORYHELPER_H
+#define ALPAKAMEMORYHELPER_H
+
+#include "AlpakaCore/alpakaConfig.h"
+#include "AlpakaCore/alpakaDevices.h"
+
+using namespace alpaka_common;
+
+namespace cms {
+  namespace alpakatools {
+
+    template <typename TData>
+    auto allocHostBuf(const Extent& extent) {
+      return alpaka::allocBuf<TData, Idx>(host, extent);
+    }
+
+    template <typename TData>
+    auto createHostView(TData* data, const Extent& extent) {
+      return alpaka::ViewPlainPtr<DevHost, TData, Dim1, Idx>(data, host, extent);
+    }
+
+    template <typename TData>
+    auto allocDeviceBuf(const Extent& extent) {
+      return alpaka::allocBuf<TData, Idx>(ALPAKA_ACCELERATOR_NAMESPACE::device, extent);
+    }
+
+    template <typename TData>
+    auto createDeviceView(const TData* data, const Extent& extent) {
+      return alpaka::ViewPlainPtr<ALPAKA_ACCELERATOR_NAMESPACE::DevAcc1, TData, Dim1, Idx>(
+          const_cast<TData*>(data),
+          ALPAKA_ACCELERATOR_NAMESPACE::device,
+          extent);  // TO DO: Obviously aweful: why no view constructor inside alpaka library with a const TData* argument?
+    }
+
+    template <typename TData>
+    auto createDeviceView(TData* data, const Extent& extent) {
+      return alpaka::ViewPlainPtr<ALPAKA_ACCELERATOR_NAMESPACE::DevAcc1, TData, Dim1, Idx>(
```

Comment:
It would be cleaner here if these functions would take the `device` as template argument, then the `ALPAKA_ACCELERATOR_NAMESPACE` would not be needed.

Reply:
Yes I had done that at first, was passing host and device as parameters.
But then I found it very redundant on call site, as in a sense we already know that allocHostBuf / allocDeviceBuf will be allocating memory on host and device respectively. And since they are global object anyway, found it nicer to just call them inside the helper functions.

Reply:
But indeed as you mentioned, some further work should be done inside the Framework, to just access evth with a ScopedContext or equivalent.